### ISSUE TITLE
OpenPegasus fails on instance reference

### ISSUE THREAD

**Issue by KSchopmeyer** [pywbem/pywbemtools#1098]
In testing with OpenPegasus, the list-destinations and list-filters table view shows 0 even when there are known subscriptions for the filters and destinations.  This may be an issue in OpenPegasus in that there appears to be a problem when an instance references request is made for the filters (See next paragraph).

An attempt to issue a reference request from pywbemcli for CIM_IndicationFilter fails in OpenPegasus.  This is part of the subscription testing because this is the request that is issued when trying to release the filter and destination references for a subscription as part of removing the subscription.

```

pywbemcli> instance references CIM_IndicationFilter.? -n root/interop --no -r Filter
Pick Instance name to process
0: //09d0c43b22c4/root/interop:CIM_IndicationFilter.CreationClassName="CIM_IndicationFilter",Name="pywbemfilter:defaultpywbemcliSubMgr:ofilter1",SystemCreationClassName="CIM_ComputerSystem",SystemName="09d0c43b22c4"
1: //09d0c43b22c4/root/interop:CIM_IndicationFilter.CreationClassName="CIM_IndicationFilter",Name="pywbemfilter:defaultpywbemcliSubMgr:pdest1",SystemCreationClassName="CIM_ComputerSystem",SystemName="09d0c43b22c4"
Input integer between 0 and 1 or Ctrl-C to exit selection: 1
Error: CIMError: 1 (CIM_ERR_FAILED): CIM_ERR_FAILED: CIM_IndicationFilter Class not supported by Interop Provider

```

Note that there is clearly a CIM_IndicationFilter class and instances in the interop namespace:

pywbemcli> instance enumerate CIM_IndicationFilter -n root/interop --no --di

```
//09d0c43b22c4/root/interop:CIM_IndicationFilter.CreationClassName="CIM_IndicationFilter",Name="pywbemfilter:defaultpywbemcliSubMgr:ofilter1",SystemCreationClassName="CIM_ComputerSystem",SystemName="09d0c43b22c4"

//09d0c43b22c4/root/interop:CIM_IndicationFilter.CreationClassName="CIM_IndicationFilter",Name="pywbemfilter:defaultpywbemcliSubMgr:pdest1",SystemCreationClassName="CIM_ComputerSystem",SystemName="09d0c43b22c4"
pywbemcli>

```

Since the correction of this issue in OpenPegasus may be problematic, The best solution may be to create an equivalent client side operation.  In the short term since the operation is only used to determine the number of subscriptions for a particular filter or destination, we propose to simply catch the error and replace the count with a ? or similar character.

This was documented in OpenPegasus with issue  References Operation not supported for CIM_Filter in interop namespace #83

**Comment by KSchopmeyer** [comment]
ACTION: Determine if the issue is actually in OpenPegasus and file an issue with OpenPegasus

**Comment by KSchopmeyer** [comment]
In another version of OpenPegasus (local last version before creation of the gityhub repo), the following occurred:

```
pywbemcli> -U no instance references CIM_IndicationFilter.? -n root/PG_interop --no
Error: instance references failed because FilterQuery not allowed with traditional References. --use-pull: False. Exception: ValueError: The 'MaxObjectCount' parameter of the WBEMConnection operation has invalid value None (must be > 0)
pywbemcli> -U yes instance references CIM_IndicationFilter.? -n root/PG_interop --no
Error: instance references failed because FilterQuery not allowed with traditional References. --use-pull: True. Exception: ValueError: The 'MaxObjectCount' parameter of the WBEMConnection operation has invalid value None (must be > 0)
pywbemcli>

```

**Comment by KSchopmeyer** [comment]
There are a couple of issued found:

1. The general option pull_max_cnt (on off, either) translates to (True, False, None) internally but thier is an issue in the pywbem Iter... methods in that they test for None for the MaxObjectCount before they determine whether the pull or non-pull operations are used and pywbemcli can deliver None if the original input is None.  Currently when the value is set into PywbemServer, the test is if there is a value in the orginal variable input and not set the resolved value if that is None.  Thus, in some cases, we pass None to the pywbem.Iter...methods which causes them to fail. The easy solution is to always be certain that the PywbemServer sets a value into its variables and that can be done by always setting the resolved value.

2. We completely ignored the issues around namespace when we try to enumerate the INSTANCENAME property so that it looks into the current namespace for a class and instances unless the namespace is specifically set into the INSTANCENAME argument.  This always works with a single namespace but when we get multiple namespaces, the whole issue gets confusing.

**Comment by andy-maier** [comment]
Is this issue also solved by PR #1101?
